taskmgr/queueservice_test.go

```diff
@@ -15,49 +19,736 @@ import (
+	var handler = func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/xml")
+		w.Write([]byte(messages))
+	}
+
+	s := httptest.NewServer(http.HandlerFunc(handler))
+	defer s.Close()
+	logger, _ := runtime.CreateLogger(os.Getenv("LOGGING_LEVEL"))
+	service := queueService{
+		Log: logger.WithField("component", "Queue Service"),
+		queues: []taskQueue{{
+			SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+			SignedPollUrl:   fmt.Sprintf("%s/tasks", s.URL),
+		}},
+	}
+
+	tasks, err := service.pollTaskUrl(&service.queues[0], 3)
+	assert.Nil(t, err, "Error should not have been returned when empty message list provided.")
+	assert.Equal(t, 2, len(tasks))
+	// quick sanity check to make sure the messages are different
+	assert.NotEqual(t, tasks[0].TaskId, tasks[1].TaskId)
+	assert.NotEqual(t, tasks[0].SignedDeleteUrl, tasks[1].SignedDeleteUrl)
+}
+
+func TestPollTaskUrlInvalidMessageTextContents(t *testing.T) {
+	// MessageText is {"abc",0} which is an invalid format when
+	// unmarshalling.
+	messages := `<?xml version="1.0" encoding="utf-8"?>
+	<QueueMessagesList>
+	  <QueueMessage>
+		<MessageId>5974b586-0df3-4e2d-ad0c-18e3892bfca3</MessageId>
+		<InsertionTime>Fri, 09 Oct 2009 21:04:30 GMT</InsertionTime>
+		<ExpirationTime>Fri, 16 Oct 2009 21:04:30 GMT</ExpirationTime>
+		<PopReceipt>YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw</PopReceipt>
+		<TimeNextVisible>Fri, 09 Oct 2009 23:29:20 GMT</TimeNextVisible>
+		<DequeueCount>1</DequeueCount>
+		<MessageText>eyJhYmMiLDB9</MessageText>
+	  </QueueMessage>
+
+	</QueueMessagesList>`
+	var handler = func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/xml")
+		w.Write([]byte(messages))
+	}
+
+	s := httptest.NewServer(http.HandlerFunc(handler))
+	defer s.Close()
+	logger, _ := runtime.CreateLogger(os.Getenv("LOGGING_LEVEL"))
+	service := queueService{
+		Log: logger.WithField("component", "Queue Service"),
+		queues: []taskQueue{{
+			SignedDeleteUrl: fmt.Sprintf("%s/delete/{{messageId}}/{{popReceipt}}", s.URL),
+			SignedPollUrl:   fmt.Sprintf("%s/tasks", s.URL),
+		}},
+	}
+
+	tasks, err := service.pollTaskUrl(&service.queues[0], 3)
+	assert.Nil(t, err, "Error should not have been raised when unmarshalling invalid MessageText")
+	assert.Equal(t, 0, len(tasks))
+}
+
+func TestPollTaskUrlInvalidMessageTextEncoding(t *testing.T) {
+	// MessageText is not a valid base64 encoded string
+	messages := `<?xml version="1.0" encoding="utf-8"?>
+	<QueueMessagesList>
+	  <QueueMessage>
+		<MessageId>5974b586-0df3-4e2d-ad0c-18e3892bfca3</MessageId>
+		<InsertionTime>Fri, 09 Oct 2009 21:04:30 GMT</InsertionTime>
+		<ExpirationTime>Fri, 16 Oct 2009 21:04:30 GMT</ExpirationTime>
+		<PopReceipt>YzQ4Yzg1MDItYTc0Ny00OWNjLTkxYTUtZGM0MDFiZDAwYzEw</PopReceipt>
+		<TimeNextVisible>Fri, 09 Oct 2009 23:29:20 GMT</TimeNextVisible>
+		<DequeueCount>1</DequeueCount>
+		<MessageText>eyJhYmMiLDB9xxxxxxx</MessageText>
```

Comment:
It wasn't really a biggy, I just thought if it had non-base64 characters in it, it would be more immediately obvious that it isn't a valid base64 string, but it doesn't really matter. E.g. `<MessageText>invalid MessageText, not base64 encoded!</MessageText>` might be clearer, but the comment already explains it is not valid base64, so it really doesn't matter.